### PR TITLE
Add emit batch arguments

### PR DESCRIPTION
**Pull request by ameech**
Adds the ability to batch emit with arguments.

Here's how I'm handling it with the current version: https://github.com/equip/queue/blob/master/src/Event.php#L31

Also, I don't have much experience with PHPSpec, so I'm not sure if the tests are up to snuff.

### CODE REVIEW

File: spec/EmitterSpec.php [hunk 1]
```diff
@@ -82,6 +82,13 @@ public function it_should_allow_you_to_emit_plain_events()
         $this->emit('event')->shouldReturnAnInstanceOf('League\Event\Event');
     }
 
+    public function it_should_allow_you_to_emit_events_with_additional_arguments()
+    {
+        $callback = function ($event, $test, $example) {};
```
Comment (ameech) [inline spec/EmitterSpec.php]: Within PHPSpec is there a way to assert that `$test == 'test'`?

Comment (frankdejonge) [inline spec/EmitterSpec.php]: @ameech you could simply create your own assertion here by throwing an exception when `$test !== 'test'`.

Comment (ameech) [inline spec/EmitterSpec.php]: Thanks @frankdejonge, that was way too obvious. Updated!

**Comment by ameech** [comment]
@frankdejonge thoughts on this change?

**Comment by frankdejonge** [comment]
@ameech I'm not sure about this. It feels weird to me that multiple events, which aren't related by anything would receive the same additional arguments. Could you explain which cases this would solve?

**Pull request closed without merging**